### DISCUSSION TITLE
popupContentView.popupCloseButton hide after once set ?

### DISCUSSION THREAD

**Discussion by divyeshdobariya** [LeoNatan/LNPopupController#587] [General] [answered]
I have one project and i am using chevron style, I want to hide that popupclosebutton on that viewcontroller's button click. Below is the code for set

```
let demoVC = MAINSTORYBOARD.instantiateViewController(identifier: "VideoPlayerVC") as! VideoPlayerVC
demoVC.popupItem.title = title
demoVC.popupItem.subtitle = subtitle
demoVC.thumbImage = image

let appearance = LNPopupBarAppearance()
appearance.marqueeScrollEnabled = true

let paragraphStyle = NSMutableParagraphStyle()
paragraphStyle.alignment = .left
paragraphStyle.lineBreakMode = .byTruncatingTail

appearance.titleTextAttributes = AttributeContainer([
    .font: BOLDFont(size: 15),
    .paragraphStyle:paragraphStyle,
    .foregroundColor: color_white_static
  ])

appearance.subtitleTextAttributes = AttributeContainer([
    .font: REGULARFont(size: 12),
    .paragraphStyle:paragraphStyle,
    .foregroundColor: color_static_music_subtitle
  ])
        
appearance.floatingBackgroundColor = TempImage.averageColor?.withAlphaComponent(0.9)


tabBarController?.popupBar.imageView.contentMode = .scaleAspectFill
tabBarController?.popupBar.imageView.layer.cornerRadius = 5
tabBarController?.popupBar.tintColor = color_white_static
tabBarController?.popupBar.progressViewStyle = .top
tabBarController?.popupBar.barStyle = .floating
tabBarController?.popupBar.standardAppearance = appearance
tabBarController?.popupContentView.popupCloseButtonStyle = .chevron
tabBarController!.presentPopupBar(withContentViewController: demoVC, animated: true, completion: nil)
```

I have to hide Popupclosebutton when click on "VideoPlayerVC" 's one button..

I tried popupContentView.popupCloseButton.isHidden = true in VideoPlayerVC, but it's not working

**Comment by LeoNatan** [comment]
Hi,

just to understand, you want the close button to be visible until some other button is tapped inside the popup content controller, then you want to hide the close button?

**Reply by divyeshdobariya** [reply]
yes, right. I have Video Player app, when I tap Fullscreen button then hide all the controls, so hide that close button

**Comment by iDevelopper** [comment, marked as answer]
In your popupContentViewController, what about

```swift
self.popupPresentationContainerViewController.popupContentView.popupCloseButton.isHidden = true
```

**Reply by LeoNatan** [reply]
I might expose that property on the content controller for easier access in the future. For now, what @iDevelopper wrote is correct.

**Reply by divyeshdobariya** [reply]
Thanks for your answer and thanks for all of the suggestions, Now it's solved. Thankssss..

**Comment by LeoNatan** [comment]
There is another option, which is probably more elegant. 

You can place the close button inside your view hierarchy, and then, if you hide that, the button will be hidden as well. 

See
https://github.com/LeoNatan/LNPopupController/blob/6ef5ba33e53b19c0c54b5034dcbb0c467ec9a704/LNPopupController/LNPopupController/UIViewController%2BLNPopupSupport.h#L79
